modules/programs/vim.nix

```diff
@@ -65,6 +65,18 @@ in {
     programs.vim = {
       enable = mkEnableOption "Vim";
 
+      source_package = mkOption {
+        type = with types; package;
+        default = pkgs.vim_configurable;
+        example = literalExample
+          "pkgs.vim_configurable.override { python = pkgs.python3; }";
+        description = ''
+          A vim package which will be installed and configured by home-manager.
+          This can either be one of the default vim packages from nixpkgs or a customized
+          package.
```

Comment:
```suggestion
          A Vim package which will be installed and configured by Home Manager.
          This can either be one of the default Vim packages from Nixpkgs or a customized
          package.
```